test/Microsoft.AspNetCore.Mvc.FunctionalTests/compiler/resources/HtmlGenerationWebSite.HtmlGeneration_Home.Index.Encoded.html

```diff
@@ -39,32 +39,32 @@
         </a>
     </div>
     <div>
-        <a href="HtmlEncode[[/Customer/Customer#fragment]]">Customer Area Customer Index</a>
+        <a href="HtmlEncode[[/Customer/HtmlGeneration_Customer#fragment]]">Customer Area Customer Index</a>
     </div>
     <div>
         <a href="/Order/List">Href Order List</a>
     </div>
     <div>
-        <a href="HtmlEncode[[/NonExistentController]]">Non-existent Controller</a>
+        <a href="">Non-existent Controller</a>
```

Comment:
There is a behavior change here, but I don't believe its a significant breaking change...in earlier version we should be generating a url which would give a 404 vs. now where it would be empty.

Reply:
Filed https://github.com/aspnet/Routing/issues/728 to track this

Reply:
As an aside - are we getting a lot of value out of these tests? This is a cross-product test of a bunch of routing features masquerading as an HTML helper functional test.

Reply:
> are we getting a lot of value out of these tests?

Actually, these tests are about more than routing because we cover how the helpers encode text when going down various paths. Further, the helpers don't just call into routing.

> If that's the case, it's intended.

Why is this intended? The 404s mean something while silently going back to the original URL is a major problem in many cases.